extensions/blocks/mailchimp/mailchimp.php

```diff
@@ -78,20 +79,34 @@ function jetpack_mailchimp_block_load_assets( $attr ) {
 			)
 		);
 	}
-	$button_styles = implode( $button_styles, ';' );
+	$button_styles   = implode( $button_styles, ';' );
+	$amp_form_action = sprintf( 'https://public-api.wordpress.com/rest/v1.1/sites/%s/email_follow/amp/subscribe/', $blog_id );
+	$is_amp_request  = class_exists( 'Jetpack_AMP_Support' ) && Jetpack_AMP_Support::is_amp_request();
 
 	ob_start();
 	?>
+
 	<div class="<?php echo esc_attr( $classes ); ?>" data-blog-id="<?php echo esc_attr( $blog_id ); ?>">
 		<div class="components-placeholder">
-			<form aria-describedby="wp-block-jetpack-mailchimp_consent-text">
+			<form
+				aria-describedby="wp-block-jetpack-mailchimp_consent-text"
+				<?php if ( $is_amp_request ) : ?>
+					action-xhr="<?php echo esc_url( $amp_form_action ); ?>"
+					method="post"
+					id="mailchimp_form"
+					target="_top"
+				<?php endif; ?>
+			>
 				<p>
 					<input
 						aria-label="<?php echo esc_attr( $values['emailPlaceholder'] ); ?>"
 						placeholder="<?php echo esc_attr( $values['emailPlaceholder'] ); ?>"
 						required
 						title="<?php echo esc_attr( $values['emailPlaceholder'] ); ?>"
 						type="email"
+						<?php if ( $is_amp_request ) : ?>
+						name="email"
```

Comment:
Do we really need to have in only for amp requests? Removing `if` statement won't harm?

Reply:
Originally I did this simply to avoid altering the non-AMP markup at all, to reduce testing and regression possibility. However, this seems completely safe, and tests fine. Done in https://github.com/Automattic/jetpack/pull/12695/commits/ed5fb5abcb2c19332890cb83d5e97caa16e7ae21